db-engine/src/main/java/com/splicemachine/db/jdbc/InternalDriver.java

```diff
@@ -468,6 +468,11 @@ public static String getDatabaseName(String url, Properties info) {
         else
             dbname = url.substring(Attribute.PROTOCOL.length(), attributeStart);
 
+        // check if we have dbname or host:port/dbname:
+        int dbnameStart = dbname.indexOf('/');
```

Comment:
That's a good point, we should not allow special characters for database names.
It will not quite crash, but it will lead to very weird database names:
```
SPLICE* - 	jdbc:splice://localhost:1527/splicedb
* = current connection
splice> connect 'jdbc:splice://localhost:1527/"splicedb2/"foo;user=isi;password=pass;create=true';
ELAPSED TIME = 70 milliseconds
splice> show databases;
DATABASENAME
------------------------------------
"FOO
SPLICEDB

2 rows selected
ELAPSED TIME = 279 milliseconds
splice> connect 'jdbc:splice://localhost:1527/splicedb2/foo/bar/;user=isi;password=pass;create=true';
ELAPSED TIME = 46 milliseconds
splice> show databases;
DATABASENAME
------------------------------------
"FOO
FOO/BAR/
SPLICEDB

3 rows selected
ELAPSED TIME = 12 milliseconds
splice> connect 'jdbc:splice://localhost:1527//;user=isi;password=pass;create=true';
ELAPSED TIME = 58 milliseconds
splice> show databases;
DATABASENAME
------------------------------------

"FOO
FOO/BAR/
SPLICEDB

4 rows selected
```

I filed DB-11571 for that.